### DISCUSSION TITLE
data from the command-line

### DISCUSSION THREAD

**Discussion by ahl** [11ty/eleventy#4091] [Enhancement Queue]
**Is your feature request related to a problem? Please describe.**
I am running eleventy in development and from netlify. I'd like the ability to inject some data from the environment into my templates.

**Describe the solution you'd like**
A command line option to the `eleventy` command to either specify a file with data (json/js/etc) or specify k/v pairs on the command line.

e.g. `eleventy --data baseUrl=$DEPLOY_PRIMEURL`

**Describe alternatives you've considered**
Here's how I worked around it with javascript data in a file I called `_data/site.json`:
```javascript
module.exports = function() {
  return new Promise((resolve, reject) => {
    var url = process.env.CONTEXT == 'production' ?
      process.env.URL :
      process.env.DEPLOY_PRIME_URL;
    resolve({
      "baseUrl": url || "http://localhost:8080"
    });
  });
};
```

**Comment by Ryuno-Ki** [comment]
What would block you from using environment variables here?

**Comment by ahl** [comment]
@Ryuno-Ki I must not be understanding your question because my workaround is to use environment variables, but it seems at best inelegant.

Nothing would block me from using environment variables. My suggestion is that it would be useful to allow data overrides to be specified at the command line.

**Comment by zachleat** [comment]
Ah I’m not sure about this one, hmm. I will put it into the queue but this seems pretty limited compared to the workaround you’re using!

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!
